File: tests/workflow/producer/rust/src/producer/beacons/beacona.rs
use super::{identification, producer::Control, protocol, scope::Scope, Context};
use crate::{stat::Alias, stop, test::samples};
use clibri::server;

#[allow(unused_variables)]
pub async fn emit<E: server::Error, C: server::Control<E>>(
    beacon: &protocol::BeaconA,
    scope: &mut Scope<'_, E, C>,
) -> Result<(), String> {
    if !samples::beacon_a::equal(beacon.clone()) {
        stop!("BeaconA isn't equal to sample");
    }
    scope
        .context
        .inc_stat(scope.identification.uuid(), Alias::BeaconA);
    scope
        .context
        .check_beacons(scope.identification.uuid(), scope.control)
        .await;
    Ok(())
}

File: tests/workflow/producer/typescript/src/producer/beacons/beacona.ts
import {
	Context,
	Producer,
	Identification,
	Filter,
	Protocol,
} from "../implementation/beacons";
import { Scope } from "../implementation/scope";
import { Alias } from "../../stat";

export function emit(beacon: Protocol.BeaconA, scope: Scope): Promise<void> {
	const stat = scope.context.getStat(scope.consumer.uuid());
	stat.case(Alias.BeaconA);
	scope.context.checkBeacons(scope.consumer.uuid(), scope.producer);
	return Promise.resolve();
}
